### PR TITLE
linux: disable NTFS_FS, enable NTFS3_LZX_XPRESS and NTFS3_FS_POSIX_ACL

### PR DESCRIPTION
**Pull request by SuperSamus**
###### Description of changes

Disables the old read-only NTFS driver, and enable the [recommended options](https://github.com/torvalds/linux/blob/master/fs/ntfs3/Kconfig) for the new NTFS3 driver.
[Arch](https://github.com/archlinux/svntogit-packages/blob/packages/linux/trunk/config#L9959) and [Fedora](https://gitlab.com/cki-project/kernel-ark/-/tree/os-build/redhat/configs/fedora/generic) use the same configuration.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fortuneteller2k** [comment]
Please target the `staging` branch.

**Comment by lovesegfault** [comment]
> Please target the `staging` branch.

You don't need staging for changes to the kernel.

File: pkgs/os-specific/linux/kernel/common-config.nix [hunk 1]
```diff
@@ -394,6 +394,10 @@ let
       EXT4_FS_SECURITY  = yes;
       EXT4_ENCRYPTION   = option yes;
 
+      NTFS_FS = no;
```
Comment (lovesegfault) [inline pkgs/os-specific/linux/kernel/common-config.nix:397]:
```suggestion
      NTFS_FS = no;
      NTFS3_FS = module;
```

Comment (SuperSamus) [reply pkgs/os-specific/linux/kernel/common-config.nix:397]: It's already implied for `NTFS3_FS` to be a module (same for every other filesystem there).

File: pkgs/os-specific/linux/kernel/common-config.nix [hunk 2]
```diff
@@ -394,6 +394,10 @@ let
...
+      NTFS3_LZX_XPRESS = yes;
+      NTFS3_FS_POSIX_ACL = yes;
```
Comment (Atemu) [inline pkgs/os-specific/linux/kernel/common-config.nix:399]: Older kernels don't have this option, do they?

**Comment by SuperSamus** [comment]
Is there anything else to do for this to be merged?

**Comment by lovesegfault** [comment]
Other than the rebase, no, I can merge it once it's done.

**Comment by lovesegfault** [comment]
Closing to avoid spamming everyone. Please reopen this PR targeting master, and not staging, as is customary for kernel work.

**Pull request closed without merging**